### PR TITLE
Predict api cicd

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
# Feature

**_WHAT DOES THIS PR DO?_**
- [x] Predict staging api service CI/CD
- [x] Predict production api service CI/CD

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
- N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
[predict-api-cicd](https://github.com/airqo-platform/AirQo-api/tree/predict-api-cicd)

**_HOW DO I TEST OUT THIS PR?_**
- N/A

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
- N/A

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- N/A

**_ARE THERE ANY RELATED PRs?_**
- N/A

### CODE REVIEW

**Review by danielogen — CHANGES_REQUESTED** [review]
Great stuff!!... I can see some `package.json` files edited but they do not relate to `predict-api`.  Is it possible to change??

**Comment by 123MwanjeMike** [comment]
> Great stuff!!... I can see some `package.json` files edited but they do not relate to `predict-api`. Is it possible to change??

Thanks @danielogen. Let me make the necessary changes.

**Review by danielogen — APPROVED** [review]
LGTM

File: .github/workflows/deploy-apis-to-production.yml [hunk 1]
```diff
@@ -736,3 +736,38 @@ jobs:
...
+          method: kubeconfig
+          kubeconfig: ${{ secrets.K8S_CONFIG_STAGE }}
+      
+      - name: Build and Push Docker Image
+        run: |
+          cd src/predict/
+          docker build --target=production --tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prediction-api:${{ needs.image-tag.outputs.build_id }} .
+          docker tag ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prediction-api:${{ needs.image-tag.outputs.build_id }} ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prediction-api:latest
+          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prediction-api:${{ needs.image-tag.outputs.build_id }}
+          docker push ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prediction-api:latest
```
Comment (NoahNsimbe) [inline .github/workflows/deploy-apis-to-production.yml:768]: Thanks @123MwanjeMike 
For future builds, we might utilise the Github Action **[Build and push Docker images](https://github.com/marketplace/actions/build-and-push-docker-images)** with Buildx enabled

```
       -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
       -
        name: Build and push
        uses: docker/build-push-action@v2
        with:
          push: true
          target: production
          context: src/predict/
          tags: ${{ secrets.REGISTRY_URL }}/${{ secrets.PROJECT_ID }}/airqo-prediction-api:${{ needs.image-tag.outputs.build_id }}
         
```
**References** : https://github.com/marketplace/actions/build-and-push-docker-images#customizing

**Review by NoahNsimbe — APPROVED** [review]
LGTM

File: .github/workflows/deploy-apis-to-production.yml (hunk 1 above)
Comment (123MwanjeMike) [reply .github/workflows/deploy-apis-to-production.yml:768]: Noted with thanks

**Pull request merged**